### DISCUSSION TITLE
Can't make ESP32C3 go into flash mode

### DISCUSSION THREAD

**Discussion by vfranchi** [arendst/Tasmota#20669] [Support request] [answered]
I bought a smart IR device that I want to flash Tasmota into it. I've seen other smart IR being flashed and I thought I'd be able to do it with this one. I've googled a bit to find which esp chip it uses, and I ended up on this doc: https://templates.blakadder.com/ESP8685-WROOM-06.html

I'm using an ESP8266 as my USB to UART converter, I've soldered the jumpers, checked for shorts. I can't seem to put it into flash mode. I can read the logs from the Tuya firmware on it, I believe I got the right ESP model on it. Can anyone in here please help me find out what I'm doing wrong?

![esp32c3](https://github.com/arendst/Tasmota/assets/2249360/4425da00-165b-46b1-979c-7f92acf6d312)
![20240205_090620 - Copia](https://github.com/arendst/Tasmota/assets/2249360/6d8e93bc-003b-471d-9578-7b4f3e3574ec)
![20240205_090608 - Copia](https://github.com/arendst/Tasmota/assets/2249360/97cd433f-aad8-4a2e-aaf1-c4bf3f17feb7)

Running `esptool.py chip_id` just fails as not connected
```
esptool.py v4.5.1
Found 1 serial ports
Serial port COM6
Connecting......................................
COM6 failed to connect: Failed to connect to Espressif device: No serial data received.
For troubleshooting steps visit: https://docs.espressif.com/projects/esptool/en/latest/troubleshooting.html

A fatal error occurred: Could not connect to an Espressif device on any of the 1 available serial ports.
```

Logs from serial monitor:
[esp32c3-log.txt](https://github.com/arendst/Tasmota/files/14165364/esp32c3-log.txt)

**Comment by sfromis** [comment, marked as answer]
To use a dev board as USB/serial adapter, see the NodeMCU paragraph here:
https://tasmota.github.io/docs/Getting-Started/#serial-programmer

Not sure why you talk about ESP32-C3 when your images show a CBU module which is another MCU type, the BK7231 (as visible in the log you posted) which is not compatible with Tasmota.

While the [module you found](https://templates.blakadder.com/ESP8685-WROOM-06.html) has the same form factor as the CBU, and is good as a replacement, it is very different internally.

**Comment by vfranchi** [comment]
> To use a dev board as USB/serial adapter, see the NodeMCU paragraph here:
https://tasmota.github.io/docs/Getting-Started/#serial-programmer

Yep, I'm using a Lolin NodeMCU as my serial adapter by grounding the EN pin from it and connecting TX -> TX, RX -> RX

> Not sure why you talk about ESP32-C3 when your images show a CBU module which is another MCU type, the BK7231 (as visible in the log you posted) which is not compatible with Tasmota.

I've googled `esp cbu tasmota` and ended up on that doc page, I thought I had the right module due to pure ignorance. Thanks for the clarification. 

My goal was to make the hardware run 100% locally through mqtt using Tasmota. Now, I'll try to learn more about this module and see if there is an alternative for me. Thanks again.

**Reply by sfromis** [reply]
What I'd be inclined to do would be to replace the CBU module by the one you found ....

**Reply by vfranchi** [reply]
I came across an Open Source Firmware for BK7231 chips: https://github.com/openshwprojects/OpenBK7231T_App

And I found the datasheet for the CBU module I have: https://developer.tuya.com/en/docs/iot/cbu-module-datasheet?id=Ka07pykl5dk4u (hopefully I'm looking at the right place this time)

I'll try that route first before venturing myself on replacing the module. Not confident on my soldering skills just yet haha
